src/annotation_processor.md

```diff
@@ -0,0 +1,149 @@
+---
+layout: guide
+title: PDL Schema
+permalink: /annotation_processor
+excerpt: Documentation of using annotation processor in Rest.Li
+---
+
+# Annotation Processor
+This sections introduces a generic schema annotation processing tool in Rest.Li. Before reading this, it is recommended to be familiar with the concept of PathSpec in Rest.Li. [Insert link here]
+
+## Annotating Rest.Li Schemas
+In Rest.Li, schema writers are able to add annotations to schema fields or the schema itself.
+
+**Example**: annotate on the schema fields
+```pdl
+record UserPersonallyIdentifiableInformation {
+  @data_classification = "MEDIUM"
+  firstName: string,
+  @data_classification = "MEDIUM"
+  lastName: string,
+  @data_classification = "LOW"
+  userId: long,
+  @data_classification = "HIGH"
+  socialSecurityNumber: string,
+}
+```
+
+**Example**: annotate directly on the schema
+```pdl
+@data_classification = "HIGH"
+record UserPersonallyIdentifiableInformation {
+  firstName: string,
+  lastName: string,
+  userId: long,
+  socialSecurityNumber: string,
+}
+```
+Note that these annotations are on the schema level.
+
+The support that Rest.Li provided for processing annotation in Rest.Li schema has been quite limited. These annotations in above examples are later stored as a `DataSchema`'s class attribute called "property". Just as the example shows, both field and the DataSchema can have this "property". We let the user chooses where they want to have the schema annotated and how they want to interpret. There were no preferrabled way regarding how annotation should be written, read and interpreted.
```

Comment:
Keep all text in third person, so avoid "we" "you" etc.

We let the user chooses where they want to have the schema annotated and how they want to interpret. There were no preferrabled way regarding how annotation should be written, read and interpreted.
->
The Rest.Li framework did not provide specification on how these annotations should be interpreted and if was left to the user to add logic for interpreting them.

(Written syntax and read as properties are specified by framework)

Reply:
Good advice. Thanks